tpm_futurepcr/LogEvent.py

```diff
@@ -73,8 +81,45 @@ def _read_data(self, binary_reader: BinaryReader):
         event_size = binary_reader.read_u32()
         self.data = binary_reader.read(event_size)
 
-    def next_extend_value(self, current_extend_value: bytes, hash_alg: str = "sha1") -> bytes:
-        return current_extend_value
+    def next_extend_value(self, hash_alg: TpmAlgorithm = TpmAlgorithm.SHA1) -> bytes:
+        return self.pcr_extend_values[hash_alg]
+
+    def show(self):
+        logger.verbose("\033[1mPCR %d -- Event <%s>\033[m", self.pcr_idx, self.type.name)
+        for i in hexdump(self.data, 64):
+            logger.debug(i)
+
+
+@dataclass
+class NoActionEvent(BaseEvent):
+
+    def __post_init__(self, binary_reader: BinaryReader, tpm_version: int, tcg_hdr: dict | None):
+        super().__post_init__(binary_reader, tpm_version, tcg_hdr)
+
+    def _read_data(self, binary_reader: BinaryReader):
+        event_size = binary_reader.read_u32()
+        self.data = binary_reader.read(event_size)
+
+    def next_extend_value(self, hash_alg: TpmAlgorithm = TpmAlgorithm.SHA1) -> bytes:
+        return self.pcr_extend_values[hash_alg]
+
+    def show(self):
+        logger.verbose("\033[1mPCR %d -- Event <%s>\033[m", self.pcr_idx, self.type.name)
+        for i in hexdump(self.data, 64):
+            logger.debug(i)
```

Comment:
Hmm, I don't like how every subclass has to reimplement `_read_data()` and `show()` if it's done in exactly the same way across all types – this looks exactly like something that could be inherited from the parent class, rather than being an abstract method.

Reply:
my bad, you are absolutely right